api-scripts/create-or-update-dataset.py

```diff
@@ -303,6 +304,115 @@ def _get_wgs84_bbox(config):
+                # try removing last folder from dataset_path
+                base_path = os.path.dirname(dirname)
+                shp_path = os.path.join(base_path, shp_name)
+
+                if os.path.exists(shp_path) or (
+                        requests.get(shp_path).status_code == 200):
+                    return shp_path
+
+    return False
+
+
+def _get_median_latlon_bounds(vector_path):
+    """Compute the average lat/lon and bbox of a vector dataset.
+
+    Compute the average latitude and longitude coordinates of a vector dataset
+    and determine a bounding box that encloses ~75% of the features (centroids)
+
+    Return
+        average lat, lon, and 75% bounding box
+
+    """
+    from osgeo.osr import OAMS_TRADITIONAL_GIS_ORDER
+
+    wgs84 = gdal.osr.SpatialReference()
+    wgs84.ImportFromEPSG(4326)
+    wgs84.SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER)
+
+    if vector_path.startswith("http"):  # read file from remote
+        vector_path = os.path.join("/vsicurl", vector_path)
+    driver = ogr.GetDriverByName('ESRI Shapefile')  # default
+    if vector_path.lower().endswith('.geojson'):
+        driver = ogr.GetDriverByName('GeoJSON')
+    elif vector_path.lower().endswith('.gpkg'):
+        driver = ogr.GetDriverByName('GPKG')
+
+    dataset = driver.Open(vector_path, 0)
+    if dataset is None:
+        raise RuntimeError(f"Could not open vector: {vector_path}")
+
+    layer = dataset.GetLayer()
+    source_srs = layer.GetSpatialRef()
+
+    if source_srs is None:
+        LOGGER.warning("No CRS found; assuming WGS84.")
+        coord_transform = None
+    else:
+        coord_transform = osr.CoordinateTransformation(source_srs, wgs84)
+
+    # sum_x, sum_y, count = 0.0, 0.0, 0
+    coords = []
+    for feature in layer:
+        geom = feature.GetGeometryRef()
+        if geom is None:
+            continue
+        geom_pt = geom.Centroid() if geom.GetGeometryType() != ogr.wkbPoint else geom
+        coord = (geom_pt.GetX(), geom_pt.GetY()) if geom_pt.GetGeometryType() in [ogr.wkbPoint, ogr.wkbPoint25D] else (None, None, None)
+        if coord[0] is None or coord[1] is None:
+            continue
+        if coord_transform:
+            x, y, _ = coord_transform.TransformPoint(coord[0], coord[1])
+            coord = (x, y)
+        coords.append(coord)
+
+    if not coords:
+        raise ValueError(f"No valid geometries found in {vector_path} to "
+                         "compute center lat/lon and bbox")
+
+    xs, ys = zip(*coords)
+    xs = numpy.array(xs)  # lons
+    ys = numpy.array(ys)  # lats
+
+    # Ensure that x (lon) is between -180 and 180, and y (lat) between -90 and 90
+    if numpy.min(xs) < -180 or numpy.max(x) > 180 or numpy.min(ys) < -90  or numpy.max(ys) > 90:
+        raise ValueError("Latitude/longitude outside expected range: Min x: "
+                         f"{numpy.min(xs)}, max x: {numpy.max(x)}, min y: "
+                         f"{numpy.min(ys)}, max y: {numpy.max(ys)}")
```

Comment:
Nice, failing loudly if things are malformed!  I like that this will catch cases where the X extents are in the range (0, 360), and also the case where we aren't in a geographic coordinate system.